examples/task_dynamic_foraging.py

```diff
@@ -0,0 +1,234 @@
+    )
+
+    # Auto block
+    AdvancedBlockAuto: AdvancedBlockMode = Field(..., title="Auto block mode")
+    SwitchThr: float = Field(
+        ..., title="Switch threshold for auto block", ge=0.0, le=1.0
+    )
+    PointsInARow: int = Field(
+        ..., title="Points in a row for auto block", ge=0
+    )
+
+    # Auto stop
+    MaxTrial: int = Field(..., title="Maximal number of trials", ge=0)
+    MaxTime: int = Field(..., title="Maximal session time (min)", ge=0)
+
+    RightValue_volume: float = Field(
+        3.0, title="Right reward size (uL)", ge=0.0
+    )
+    LeftValue_volume: float = Field(3.0, title="Left reward size (uL)", ge=0.0)
+
+    # Optional: Add additional validation to fields.
+    @field_validator("UncoupledReward")
+    @classmethod
+    def check_something(cls, v: int, info: ValidationInfo):
+        """Your validation code here"""
+        return v
+
+
+class DynamicForagingTask(abc.Task):
+    """
+    Example Task
+    """
+
+    name: Literal["DynamicForaging"] = "DynamicForaging"
+    description: str = Field(default="AIND dynamic foraging task")
+    version: abc.SemVerAnnotation = abc.__version__
+    # ^Use the version of your task repo package!
+
+    task_parameters: DynamicForagingParas = Field(
+        ..., description=DynamicForagingParas.__doc__
+    )
+
+
+if __name__ == "__main__":
+    # Create task, optionally add parameters
+    ex_parameters = DynamicForagingParas(
+        SubTask=DynamicForagingSubTask.C1B1,
+        # -- Essentials --
+        # p_sum = 0.8, p_ratio = [1:0]
+        BaseRewardSum=0.8,
+        RewardFamily=3,
+        RewardPairsN=1,
+        # block = [10, 20, 5]
+        BlockMin=10,
+        BlockMax=20,
+        BlockBeta=5,
+        BlockMinReward=0,
+        # Small ITI at the beginning to better engage the animal
+        ITIMin=1,
+        ITIMax=7,
+        ITIBeta=3,
+        # Add a (fixed) small delay period at the beginning
+        DelayMin=0.5,
+        DelayMax=0.5,
+        DelayBeta=0,
+        # Reward delay
+        RewardDelay=0,
+        # -- Within session automation --
+        # Auto water
+        AutoReward=True,
+        AutoWaterType=AutoWaterMode.NATURAL,
+        Unrewarded=5,
+        Ignored=5,
+        Multiplier=0.5,
+        # Auto block
+        AdvancedBlockAuto=AdvancedBlockMode.NOW,
+        SwitchThr=0.5,
+        PointsInARow=5,
+        # Auto stop; set StopIgnores to a large number at the beginning
+        MaxTrial=1000,
+        MaxTime=90,
+        StopIgnores=20000,
+        # -- Miscs --
+        ResponseTime=5,
+        RewardConsumeTime=3,  # Very long response time at the beginning
+        UncoupledReward="",  # Only valid in uncoupled task
+    )
+
+    ex_task = DynamicForagingTask(
+        version="1.0.0",
+        description=("Phase B in Han's slides "
+                     "(block = [10, 20, 5], p_sum = 0.8, p_ratio = [1:0])"),
+        task_parameters=ex_parameters,
+    )
+    print(ex_task)
+
+    import json
```

Comment:
Move 'import json' to the top of the file, this is not picked up by the linters

Reply:
done.